File: app/src/main/java/de/bornholdtlee/defaultproject/utils/NetworkUtils.java
package de.bornholdtlee.defaultproject.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkUtils {

    private Context context;

    public NetworkUtils(Context context) {
        this.context = context;
    }

    @SuppressWarnings("deprecation")
    public boolean isWifiConnected() {
        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);

        return networkInfo.isConnectedOrConnecting();
    }

    public boolean isNetworkAvailable() {
        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
        return activeNetworkInfo != null && activeNetworkInfo.isConnected();
    }
}

File: app/src/main/java/de/bornholdtlee/defaultproject/utils/NetworkUtils.kt
package de.bornholdtlee.defaultproject.utils

import android.content.Context
import android.net.ConnectivityManager

class NetworkUtils(private val context: Context) {

    var isWifiConnected: Boolean = false
        get() {
            val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val networkInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI)

            return networkInfo.isConnectedOrConnecting
        }

    var isNetworkAvailable: Boolean = false
        get() {
            val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val activeNetworkInfo = connectivityManager.activeNetworkInfo
            return activeNetworkInfo != null && activeNetworkInfo.isConnected
        }
}